src/coreclr/jit/gentree.cpp

```diff
@@ -19287,6 +19287,18 @@ bool GenTree::isContainableHWIntrinsic() const
             return true;
         }
 
+        case NI_SSE3_MoveAndDuplicate:
+        case NI_AVX_BroadcastScalarToVector128:
+        case NI_AVX2_BroadcastScalarToVector128:
+        case NI_AVX_BroadcastScalarToVector256:
+        case NI_AVX2_BroadcastScalarToVector256:
+        case NI_AVX512F_BroadcastScalarToVector512:
```

Comment:
Yes, but if you have `Avx512BW.BroadcastScalarToVector512(Vector128.CreateScalarUnsafe((byte)5))`

that is equivalent to having: `Avx512F.BroadcastScalarToVector512(Vector128.CreateScalarUnsafe(0x0505_0505))` and so can still be optimized.

This could be encountered, for example, such as `Avx512F.Add(vector1, Avx512BW.BroadcastScalarToVector512(Vector128.CreateScalarUnsafe((byte)5)).AsInt32())`

It's probably not super likely to be encountered, but various code patterns can surface it so it would be good to handle longer term

Reply:
Got it, thanks for the explanation!